frontend/packages/dev-console/src/components/pipelines/pipeline-form/PipelineForm.tsx

```diff
@@ -17,6 +17,8 @@ const PipelineForm: React.FC<PipelineFormProps> = ({ PipelineFormComponent, form
     resources: _.get(obj.spec, 'resources', []),
   };
 
+  const validationSchema = getValidationSchemaForPipelineTabs(PipelineFormComponent);
```

Comment:
cyclic dependency now.
I suggest you have have the schema passed in as a prop so the parent who controls the form, also controls the validation schema.

Reply:
Updated. Now the parent is sending form component as well as validation schema.